nim/tests/test_routes.py

```diff
@@ -0,0 +1,110 @@
+import json
+from nim.app import app
+import pytest
+
+
+@pytest.fixture
+def client_app():
+    return app.test_client()
+
+
+def assert_new_game_success(passed_client, req):
+    """
+    Asserts post to /new is 200
+    & state is not empty
+    :param passed_client: test_client
+    :param req: json to post
+    :return: bool
+    """
+    response = passed_client.post(
+        '/new',
+        content_type='application/json',
+        data=req
+    )
+    assert response.status_code == 200 and not len(json.loads(response.data)['state']) == 0
+
+
+def assert_new_game_failed(passed_client, req):
+    """
+    Asserts post tp /new gets error code 400
+     & the string 'error' exists in the response
+    :param passed_client: test_client
+    :param req: json to post
+    :return: bool
+    """
+    response = passed_client.post(
+        '/new',
+        content_type='application/json',
+        data=req
+    )
+    assert response.status_code == 400 and b'error' in response.data
+
+
+def test_index(client_app):
+    response = client_app.get('/', content_type='html/text')
+    assert response.status_code == 200
+
+
+def test_new_success(client_app):
+    assert_new_game_success(client_app,
+                            json.dumps({'min': 5, 'max': 15, 'piles': 4}))
+
+
+def test_new_omit_min_success(client_app):
+    assert_new_game_success(client_app,
+                            json.dumps({'max': 15, 'piles': 4}))
+
+
+def test_new_omit_max_success(client_app):
+    assert_new_game_success(client_app,
+                            json.dumps({'min': 5, 'piles': 4}))
+
+
+def test_new_omit_piles_success(client_app):
+    assert_new_game_success(client_app,
+                            json.dumps({'min': 5, 'max': 15}))
+
+
+def test_new_omit_all_success(client_app):
+    assert_new_game_success(client_app,
+                            json.dumps({}))
+
+
+def test_new_max_limit_fail(client_app):
+    assert_new_game_failed(client_app,
+                           json.dumps({'min': 10, 'max': 51, 'piles': 3}))
+
+
+def test_new_piles_limit_fail(client_app):
+    assert_new_game_failed(client_app,
+                           json.dumps({'min': 10, 'max': 10, 'piles': 51}))
+
+
+def test_new_all_limit_fail(client_app):
+    assert_new_game_failed(client_app,
+                           json.dumps({'min': 51, 'max': 51, 'piles': 51}))
+
+
+def test_new_pile_zero_fail(client_app):
+    assert_new_game_failed(client_app,
+                           json.dumps({'min': 10, 'max': 20, 'piles': 0}))
+
+
+def test_new_all_zero_fail(client_app):
```

Comment:
This test should also probably be replaced, as far as we know `test_new_pile_zero_fail` failed so this one will fail too because `piles` is 0, thus this test doesn't give us anything.
Instead checking what happens if you just pass 0 for `min` or `max` would give us something.

Reply:
Changed